codec/codec.go

```diff
@@ -2,30 +2,23 @@ package codec
 
 import (
 	"github.com/cosmos/cosmos-sdk/codec"
-	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/types/module"
-	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
 
-	cryptocodec "github.com/cosmos/ethermint/crypto/ethsecp256k1"
+	cryptocodec "github.com/cosmos/ethermint/crypto/codec"
 	ethermint "github.com/cosmos/ethermint/types"
 )
 
-// MakeCodec registers the necessary types and interfaces for an sdk.App. This
-// codec is provided to all the modules the application depends on.
-//
-// NOTE: This codec will be deprecated in favor of AppCodec once all modules are
-// migrated to protobuf.
-func MakeCodec(bm module.BasicManager) *codec.Codec {
-	cdc := codec.New()
-
-	bm.RegisterCodec(cdc)
-	vesting.RegisterCodec(cdc)
-	sdk.RegisterCodec(cdc)
-	cryptocodec.RegisterCodec(cdc)
-	codec.RegisterCrypto(cdc)
-	ethermint.RegisterCodec(cdc)
-	keys.RegisterCodec(cdc) // temporary. Used to register keyring.Info
+// RegisterLegacyAminoCodec registers Interfaces from types, crypto, and SDK std.
```

Comment:
can we add a description here? not sure what is happening here. the codec is used for encoding/decoding messages that ared passed, correct?